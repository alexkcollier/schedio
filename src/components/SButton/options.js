export {
  colors,
  allShadeOptions,
  colorNames,
} from '@/utils/component-options'

export const sizes = [
  'small',
  'regular',
  'large',
]

export const types = [
  'outlined',
  'text',
]
