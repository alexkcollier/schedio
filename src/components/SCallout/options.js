export const calloutTypes = ['', 'success', 'warning', 'error']
