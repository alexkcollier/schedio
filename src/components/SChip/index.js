import SChip from './SChip.vue';

export { SChip };

export default {
  install: Vue => Vue.component(SChip.name, SChip),
};
