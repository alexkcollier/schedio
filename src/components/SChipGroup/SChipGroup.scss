.chip-group {
  display: flex;
  flex-wrap: wrap;
  margin-bottom: -1 * spacing();
  margin-right: -1 * spacing();

  .chip {
    margin-bottom: spacing();
    margin-right: spacing();
  }

  @each $color-name in $button-colors {
    &--color-#{$color-name} {
      .chip {
        background-color: color($color-name);
        color: choose-contrast-color($color-name);

        .chip__close {
          box-shadow: none;

          &:active,
          &:not(:disabled):not(:active):hover {
            box-shadow: none !important;
          }
        }
      }
    }
  }
}
