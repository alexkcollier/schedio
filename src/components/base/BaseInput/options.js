export const allowed = ['email', 'text', 'password', 'url', 'search', ''].sort()
