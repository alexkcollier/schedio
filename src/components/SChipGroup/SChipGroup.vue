<template>
  <Component
    :is="tag"
    class="chip-group"
    :class="{
      [`chip-group--color-${color}`]: color,
    }"
  >
    <slot />
  </Component>
</template>

<script>
import { colors } from '@/components/SChip/options'

export default {
  name: 'SChipGroup',

  props: {
    color: {
      type: String,
      default: '',
      validator: (val) => {
        return (
          !val ||
          colors.includes(val) ||
          console.error(`\`color\` must be one of ${colors.join(', ')}`)
        )
      },
    },
    tag: {
      type: String,
      default: 'div',
    },
  },
}
</script>
