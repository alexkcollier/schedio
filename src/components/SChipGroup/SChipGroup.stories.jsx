import { SChip } from '@/components/SChip'
import { SChipGroup } from '@/components/SChipGroup'
import { SHeading } from '@/components/SHeading'
import PropList from '@@/docs/components/PropList'
import { select } from '@storybook/addon-knobs'
import { colors } from '../SChip/options'

export default {
  title: 'Components|Chips/SChipGroup',
  component: SChipGroup,
  parameters: {
    jest: 'SChipGroup',
  },
}

export const ChipGroup = () => ({
  props: {
    props: {
      default: {
        color: select('color', ['', ...colors], '', 'Optional Props'),
      },
    },
  },
  render (h) {
    const { props } = this.$props

    return (
      <div>
        <SHeading>Chip Group</SHeading>
        <p>
          Chip groups display chips nicely. The chip group's color overrides an individual chip's
          color.
        </p>

        <SHeading level="2">Example</SHeading>
        <SChipGroup {...{ props }}>
          <SChip>Chip 1</SChip>
          <SChip>Chip 2</SChip>
          <SChip>Chip 3</SChip>
        </SChipGroup>

        {/* This doesn't work if props are only generated through a mixin */}
        {SChipGroup.props && <PropList component={SChipGroup} />}
      </div>
    )
  },
})

ChipGroup.story = {
  name: 'ChipGroup',
}
