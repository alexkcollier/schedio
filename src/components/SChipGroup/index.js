import SChipGroup from './SChipGroup.vue';

export { SChipGroup };

export default {
  install: Vue => Vue.component(SChipGroup.name, SChipGroup),
};
