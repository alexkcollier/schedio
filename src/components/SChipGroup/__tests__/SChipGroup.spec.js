import { shallowMount, mount } from '@vue/test-utils'
import SChipGroup from '@/components/SChipGroup/SChipGroup.vue'
import { axe, toHaveNoViolations } from 'jest-axe'
import { colors } from '@/components/SChip/options'

expect.extend(toHaveNoViolations)

describe('SChipGroup.vue', () => {
  const errorSpy = jest.spyOn(global.console, 'error').mockImplementation(() => { })
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(SChipGroup)
  })

  afterEach(() => {
    jest.resetAllMocks()
  })

  it('renders correctly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
    expect(await axe(wrapper.html())).toHaveNoViolations()
  })

  it('has a slot', () => {
    const slotContent = 'test content'
    wrapper = mount(SChipGroup, {
      slots: {
        default: slotContent,
      },
    })

    expect(errorSpy).not.toHaveBeenCalled()
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.text()).toContain(slotContent)
  })

  it.each(colors)('can have color %s', async (color) => {
    const colorWrapper = shallowMount(SChipGroup, { propsData: { color } })

    expect(errorSpy).not.toHaveBeenCalled()
    expect(colorWrapper.html()).toMatchSnapshot()
    expect(await axe(colorWrapper.html())).toHaveNoViolations()
  })

  it('validates colors', () => {
    shallowMount(SChipGroup, { propsData: { color: 'Not a color' } })
    expect(errorSpy).toHaveBeenCalledTimes(2)
  })
})
