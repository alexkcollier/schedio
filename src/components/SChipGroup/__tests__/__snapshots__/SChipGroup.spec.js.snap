// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SChipGroup.vue can have color gold 1`] = `<div class="chip-group chip-group--color-gold"></div>`;

exports[`SChipGroup.vue can have color green 1`] = `<div class="chip-group chip-group--color-green"></div>`;

exports[`SChipGroup.vue can have color grey 1`] = `<div class="chip-group chip-group--color-grey"></div>`;

exports[`SChipGroup.vue can have color ice 1`] = `<div class="chip-group chip-group--color-ice"></div>`;

exports[`SChipGroup.vue can have color magenta 1`] = `<div class="chip-group chip-group--color-magenta"></div>`;

exports[`SChipGroup.vue can have color night 1`] = `<div class="chip-group chip-group--color-night"></div>`;

exports[`SChipGroup.vue can have color red 1`] = `<div class="chip-group chip-group--color-red"></div>`;

exports[`SChipGroup.vue can have color spartan_blue 1`] = `<div class="chip-group chip-group--color-spartan_blue"></div>`;

exports[`SChipGroup.vue can have color turquoise 1`] = `<div class="chip-group chip-group--color-turquoise"></div>`;

exports[`SChipGroup.vue can have color white 1`] = `<div class="chip-group chip-group--color-white"></div>`;

exports[`SChipGroup.vue has a slot 1`] = `<div class="chip-group">test content</div>`;

exports[`SChipGroup.vue renders correctly 1`] = `<div class="chip-group"></div>`;
